src/variables.jl

```diff
@@ -790,3 +790,15 @@ function all_variables(model::Model)
         model, MOI.ListOfVariableIndices())::Vector{MOI.VariableIndex}
     return VariableRef[VariableRef(model, idx) for idx in all_indices]
 end
+
+"""
+    dual(vref::VariableRef)
+
+Returns a proper error saying that the user should refer to UpperBoundRef,
+LowerBoundRef or FixRef to get the reduced dual from a constraint.
+"""
+function dual(vref::VariableRef)
+    error("To query the dual variables associated with a variable bound, first " *
+    "obtain a constraint reference using one of UpperBoundRef, LowerBoundRef, " *
+    "or FixRef, and then call dual on the returned constraint reference.")
```

Comment:
These lines should be indented to line up with the `"` on the first line. You could also add an example:
```
For example, if `x <= 1`, instead of `dual(x)`, call `dual(UpperBoundRef(x))`.
```